cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectAssemblyOperatorTest.java

```diff
@@ -81,6 +95,51 @@ public static void after() {
         vertx.close();
     }
 
+    private ResourceOperatorSupplier supplierWithMocks() {
+        RouteOperator routeOps = openShift ? mock(RouteOperator.class) : null;
+
+        ResourceOperatorSupplier supplier = new ResourceOperatorSupplier(
+                mock(ServiceOperator.class), routeOps, mock(ZookeeperSetOperator.class),
+                mock(KafkaSetOperator.class), mock(ConfigMapOperator.class), mock(SecretOperator.class),
+                mock(PvcOperator.class), mock(DeploymentOperator.class),
+                mock(ServiceAccountOperator.class), mock(RoleBindingOperator.class), mock(ClusterRoleBindingOperator.class),
+                mock(NetworkPolicyOperator.class), mock(PodDisruptionBudgetOperator.class), mock(CrdOperator.class));
+        when(supplier.serviceAccountOperator.reconcile(anyString(), anyString(), any())).thenReturn(Future.succeededFuture());
+        when(supplier.roleBindingOperator.reconcile(anyString(), any())).thenReturn(Future.succeededFuture());
+        when(supplier.clusterRoleBindingOperator.reconcile(anyString(), any())).thenReturn(Future.succeededFuture());
+
+        if (openShift) {
+            when(supplier.routeOperations.reconcile(anyString(), anyString(), any())).thenReturn(Future.succeededFuture());
+            when(supplier.routeOperations.hasAddress(anyString(), anyString(), anyLong(), anyLong())).thenReturn(Future.succeededFuture());
+            when(supplier.routeOperations.get(anyString(), anyString())).thenAnswer(i -> {
+                return new RouteBuilder()
+                        .withNewStatus()
+                        .addNewIngress()
+                        .withHost(i.getArgument(0) + "." + i.getArgument(1) + ".mydomain.com")
+                        .endIngress()
+                        .endStatus()
+                        .build();
+            });
+        }
+
+        when(supplier.serviceOperations.hasIngressAddress(anyString(), anyString(), anyLong(), anyLong())).thenReturn(Future.succeededFuture());
+        when(supplier.serviceOperations.hasNodePort(anyString(), anyString(), anyLong(), anyLong())).thenReturn(Future.succeededFuture());
+        when(supplier.serviceOperations.get(anyString(), anyString())).thenAnswer(i -> {
+            return new ServiceBuilder()
+                    .withNewStatus()
+                    .withNewLoadBalancer()
+                    .withIngress(new LoadBalancerIngressBuilder().withHostname(i.getArgument(0) + "." + i.getArgument(1) + ".mydomain.com").build())
+                    .endLoadBalancer()
+                    .endStatus()
+                    .withNewSpec()
+                    .withPorts(new ServicePortBuilder().withNodePort(31245).build())
+                    .endSpec()
+                    .build();
+        });
+
+        return supplier;
```

Comment:
This seems to be a copy of the code in `KafkaConnectAssemblyOperatorMockTest`. Can you factor it into a common method somewhere? ResourceUtils might be a good place.